include/swift/AST/Requirement.h

```diff
@@ -43,8 +46,11 @@ enum class RequirementKind : unsigned {
 /// \brief A single requirement placed on the type parameters (or associated
 /// types thereof) of a
 class Requirement {
-  llvm::PointerIntPair<Type, 2, RequirementKind> FirstTypeAndKind;
-  Type SecondType;
+  llvm::PointerIntPair<Type, 3, RequirementKind> FirstTypeAndKind;
+  union {
```

Comment:
It's fine to leave it as-is, if you add a comment explaining that it must match the tag.